aip/general/0193.md

```diff
@@ -34,401 +34,235 @@ is necessary, you **should** provide a link where a reader can get more
-
-    > The Book, "The Great Gatsby", is unavailable at the Library,
-    > "Garfield East". It is expected to be available again on 2199-05-13.
-
-  - In contrast, using either of the following reasons is not
-    recommended: `THE_BOOK_YOU_WANT_IS_NOT_AVAILABLE`, `ERROR`. And,
-    using either of the following reasons breaches the required
-    formatting and is not allowed: `librariesAreGreat`, `noBooks`.
-
-- `domain string`
-  - The logical grouping to which the `reason` belongs. The error domain
-    is typically the registered service name of the tool or product that
-    generated the error. The domain must be a globally unique value.
-  - *Example*:<br>`pubsub.googleapis.com`
-
-- `metadata map`
-  - Additional structured details about this error, which **should**
-    provide important context for customers to identify resolution
-    steps.  Keys **should** match `/[a-z][a-zA-Z0-9-_]+/`, and be
-    limited to 64 characters in length. When identifying the current
-    value of an exceeded limit, the units **should** be contained in the
-    key, not the value.
-  - *Example*:
-    ```
-    "vmType": "e2-medium",
-    "attachment": "local-ssd=3,nvidia-t4=2",
-    "zone": "us-east1-a"
-    ```
-  - For guidance on using the metadata map, see [Dynamic
-    Variables](#dynamic-variables).
-
-##### Dynamic variables
-
-The best, actionable error messages include dynamic segments. These
-variable parts of the message are specific to a particular request.
-Without such context, it is unlikely that the message will be fully
-actionable by the user.
-
-This practice is critical so that machine actors do not need to rely on
-`LocalizedMessage.message`, which is subject to change and is not part
-of the API contract.
-
-Consider the following example:
-
-> The Book, "The Great Gatsby", is unavailable at the Library, "Garfield
-> East".  It is expected to be available again on 2199-05-13.
-
-The preceding error message is made actionable by the context, both
-originating from the request, the title of the Book, the name of the
-Library, and by the information that is known only by the service, that
-is, the expected return date of the Book.
-
-All dynamic variables found in error messages **must** also be present
-in the `map<string, string>`, `ErrorInfo.metadata` (found on the
-*required* `ErrorInfo`). For example, the `metadata` map for the sample
-error message above will include *at least* the following key/value
-pairs:
+The `reason` field is a short snake_case description of the cause of the
+error. Error reasons are unique within a particular domain of errors.
+The reason **must** be at most 63 characters and match a regular expression of
+`[A-Z][A-Z0-9_]+[A-Z0-9]`. (This is UPPER_SNAKE_CASE, without leading
+or trailing underscores, and without leading digits.)
 
-```
-bookTitle: "The Great Gatsby"
-library: "Garfield East"
-expectedReturnDate: "2199-05-13"
-```
+The reason should be terse, but meaningful enough for a human reader to
```

Comment:
```suggestion
The reason **should** be terse, but meaningful enough for a human reader to
```

Nit: Is this a requirement word or should we use a different one (or should we not fuss about use of requirement wrods in a non-requirement context?)